django/db/models/fields/files.py

```diff
@@ -312,6 +313,8 @@ def get_prep_value(self, value):
 
     def pre_save(self, model_instance, add):
         file = super().pre_save(model_instance, add)
+        if file.name is None and getattr(file, "_file") is not None:
+            raise FieldError("File name cannot be empty.")
```

Comment:
I wonder if we could provide a slightly more informative error message here. 🤔

```suggestion
            raise FieldError("File for FIELD NAME must have the name attribute specified to be saved.")
```

That's just a first pass, but it tells me what I need to look for at least. \

I wonder if from PY311 we could use [`add_note`](https://docs.python.org/3.12/library/exceptions.html#BaseException.add_note) to add a hint. (`ContentFile` should be passed the `name` parameter in order to be saved on a model field.)

Reply:
Great idea to use `add_note()`. I think we’d only want to mention `ContentFile` if explicitly used, so:

```python
exc = FieldError(f"File for {self.name} must have the name attribute specified to be saved.")
if PY311 and isinstance(file, ContentFile):
    exc.add_note("Pass a 'name' argument to ContentFile.")
raise exc
```

(I reworded the message to use active voice.)